K2Field.K2NE.ServiceBroker/ServiceObjects/Client API/OutOfOfficeClientSO.cs

```diff
@@ -228,6 +244,73 @@ private void AddOutOfOffice()
         }
 
         /// <summary>
+        /// Remove OOF share for current user
+        /// </summary>
+        [Obsolete("Method is not working, please use RemoveAllShares instead.")]
+        private void RemoveOutOfOffice()
+        {
+
+            string destinationUser = base.GetStringProperty(Constants.SOProperties.OutOfOffice.DestinationUser);
+
+            ServiceObject serviceObject = base.ServiceBroker.Service.ServiceObjects[0];
+            serviceObject.Properties.InitResultTable();
+            DataTable results = base.ServiceBroker.ServicePackage.ResultTable;
+
+
+            using (Connection k2Con = new Connection())
+            {
+                k2Con.Open(base.K2ClientConnectionSetup);
+
+                WorklistShares wsColl = k2Con.GetCurrentSharingSettings(ShareType.OOF);
+
+                //  Throw error if multiple configurations (WorklistShare objects) detected, as this method cannot support that 
+                if (wsColl.Count != 1)
+                {
+                    throw new ApplicationException(Constants.ErrorMessages.MultipleOOFConfigurations);
+                }
+
+
+                //  If configuration exist already, remove user and re-share 
+                else if (wsColl.Count == 1)
```

Comment:
It has to be done this way - the API allows to have multiple items in this collection, but the UI works properly only with first share in WSColl.